### ISSUE TITLE
Sample code fails with TypeError

### ISSUE THREAD

**Issue by JSchoeck** [paulknysh/blackbox#29]
The sample code from the readme fails on my Python 3.10.5 install.
```
result = bb.minimize(f=fun, # given function
    domain=[[-5, 5], [-5, 5]], # ranges of each parameter
    budget=20, # total number of function calls available
    batch=4 # number of calls that will be evaluated in parallel
)
print(result["best_x"])
print(result["best_f"])
```
Leads to:
```
INFO     evaluating batch 1/5 (samples 1..4/20) 10-06 10:54:00
INFO     evaluating batch 2/5 (samples 5..8/20) 10-06 10:54:04
INFO     evaluating batch 3/5 (samples 9..12/20) 10-06 10:54:09
Traceback (most recent call last):
  File "Optimierung_Logistikkosten_BlackBox.py", line 273, in <module>
    main()
  File "Optimierung_Logistikkosten_BlackBox.py", line 208, in main
    result = bb.minimize(f=fun, # given function
  File "\blackbox\blackbox.py", line 86, in minimize
    v1 = compute_volume_unit_ball(d)
  File "\blackbox\blackbox.py", line 165, in compute_volume_unit_ball
    v1 = np.pi ** (d / 2) / np.math.factorial(d / 2)
TypeError: 'float' object cannot be interpreted as an integer
```
Note that I have redacted the path information.

**Comment by paulknysh** [comment]
@JSchoeck thanks, I actually didn't know it errors (seems like in Python 3.10 and above), thought it's just a warning.

Do you want to open a small PR with a fix? (if not, I can do that)

**Comment by solbjorg** [comment]
This is still a problem, at least with the version from PyPi (which seems to be the same in this case except with different formatting). I'm happy to provide a PR to fix it. (The PR that was closed last year only fixes half the issue.)

**Comment by solbjorg** [comment]
I went ahead and made a PR. Note that pyproject.toml as it stands also specifies that it will _only_ work with Python 3.9.7 which is maybe not what was intended? I can open a separate PR for that if you'd like, I've already made a branch in my fork that changes this to require _at least_ 3.9.7. Great work by the way, this library is incredibly useful :)

**Comment by paulknysh** [comment]
Thanks @solbjorg, just approved your PR. And yes, feel free to open a PR to allow newer versions of Python too, not sure why I locked it to 3.9 only.
